Automattic/pocket-casts-android
# Add Plabyack 2024 entry points

## Description

This PR introduces entry points for the Playback 2024 feature, including both the bottom sheet and the profile card.

Designs: lH66LwxxgG8btQ8NrM0ldx-fi-3089_19654

I added an animation effect to the bottom sheet due to the need to dynamically compute the font size based on the available dimensions, which can take some time. While we have an `AutoResizeText` component, it doesn't meet my requirements because the text size must be calculated to fit either vertically on regular devices or horizontally on smaller devices. Additionally, I had to compute padding dynamically, as the designs "cheat" by ignoring the descent of the next line in the Humane font.

Currently, tapping on the entry points navigates to an empty screen.

I've removed most of the old code. Although I may reuse some parts of it, I had to remove it because it was too challenging to develop PB24 alongside the existing PB23 code. I believe it will be easier for both myself and reviewers if I reintroduce necessary parts as they are needed.

## Testing Instructions

> [!NOTE]  
> You can reset the Playback 2024 bottom sheet prompt you can go to developer settings and tap "Reset modal/profile badge".

1. Sign in.
2. Notice the bottom sheet prompt and verify it with the designs.
3. Go to the profile tab.
4. Notice the profile card and verify it with the designs.

## Screenshots or Screencast 

### Sheet themes

| Light | Dark | Rose | Indigo | Extra Dark |
| - | - | - | - | - |
| ![light](https://github.com/user-attachments/assets/ffa13362-72b6-4899-a07a-c32f8f3340a7) | ![dark](https://github.com/user-attachments/assets/1953dba0-29b7-4221-ac47-e85ca66e93ea) | ![rose](https://github.com/user-attachments/assets/c501f3f7-d5cc-4a30-915c-4deb7e9a1154) | ![indigo](https://github.com/user-attachments/assets/8a790bf5-5b1d-4afc-9710-90b59ec03c7a) | ![extra-dark](https://github.com/user-attachments/assets/ad7d4d55-2369-48d0-8507-e93d633c6f74) |

| Dark Contrast | Light Contrast | Electricity | Classic | Radioactivity |
| - | - | - | - | - |
| ![dark-contrast](https://github.com/user-attachments/assets/2216d63b-0a9d-4328-96a6-52f11471c4de) | ![light-contrast](https://github.com/user-attachments/assets/018bbed5-e576-430f-8b2b-9e3fa21f117c) | ![electricity](https://github.com/user-attachments/assets/ad0b2073-7956-448f-bf90-8976b6dbeb99) | ![classic](https://github.com/user-attachments/assets/07b9e049-e563-4506-84bb-12b78640d10f) | ![radioactivity](https://github.com/user-attachments/assets/97bea591-07df-4736-8ad4-e559d7cdf63f) |

### Sheet font size

| Regular | Small | Large |
| - | - | - |
| ![font-regular](https://github.com/user-attachments/assets/a4585731-9944-4278-9e60-1046c6beb8f4) | ![font-small](https://github.com/user-attachments/assets/c4ca1b60-43a1-4885-928e-bb10688ab82c) | ![font-large](https://github.com/user-attachments/assets/46b4727d-f602-420e-8c15-e02c70f43dc4) |

### Sheet device size 

| Regular | Small | Large |
| - | - | - |
| ![size-regular](https://github.com/user-attachments/assets/79edece7-fa62-400e-ac39-309b3d62b94f) | ![size-small](https://github.com/user-attachments/assets/6ddb46f5-7298-4fb4-bbd1-fe4d2c6f87c9) | ![size-tablet](https://github.com/user-attachments/assets/fb89d38e-66ed-4161-bab3-945a3507e9ed) |

### Profile card font size

| Regular | Small | Large |
| - | - | - |
| ![profile-regular](https://github.com/user-attachments/assets/c55cd38b-92fc-44c4-b029-2c569f311f67) | ![profile-small](https://github.com/user-attachments/assets/f124aef6-3688-402b-8c1e-8c5d7261bd39) | ![profile-large](https://github.com/user-attachments/assets/781092ff-139c-4152-ae1e-25b85bf51dbb) |
## Checklist
 
#### I have tested any UI changes...
